hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSInputStream.java

```diff
@@ -197,6 +197,15 @@ private void clearLocalDeadNodes() {
     deadNodes.clear();
   }
 
+  /**
+   * Clear list of ignored nodes used for hedged reads.
+   */
+  private void clearIgnoredNodes(Collection<DatanodeInfo> ignoredNodes) {
```

Comment:
Does it make more sense to clean up ignored / dead in the same function so we don't need to worry about calling both?

Reply:
I could add a `clearSkippedNodes` and then clear both dead and ignored in there, but that might be confusing as it sounds like theres another node type/list. I don't think `clearLocalDeadNodes` should also clear ignoredNodes because thats a bit miselading. The deadnodes and ignore nodes are handled differently, so i don't think its crazy to keep them separate and clear. 

(like i said before) i would add a method that clears both dead and ignore, but concerned it may be confusing. lmk what you think

Reply:
I'd personally err on the side of "slightly confusing name with documentation but ensure it always happens."

`clearCachedNodeState`?

Reply:
sounds good, to be clear this is what im planning

```
  private void clearCachedNodeState(Collection<DatanodeInfo> ignoredNodes) {
    clearLocalDeadNodes(); //2nd option is to remove only nodes[blockId]
    clearIgnoredNodes(ignoredNodes);
  }
 ```

Reply:
fixed